subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -135,13 +136,13 @@ public DefaultTransformer(
         this.cacheable = cacheable;
     }
 
-    public static void validateInputFileNormalizer(String propertyName, @Nullable Class<? extends FileNormalizer> normalizer, boolean cacheable, WorkValidationContext workValidationContext) {
+    public static void validateInputFileNormalizer(String propertyName, @Nullable Class<? extends FileNormalizer> normalizer, boolean cacheable, TypeValidationContext validationContext) {
         if (cacheable) {
             if (normalizer == AbsolutePathInputNormalizer.class) {
-                workValidationContext.visitWarning(null, propertyName, "is declared to be sensitive to absolute paths. This is not allowed for cacheable transforms");
-            }
-            if (normalizer == null) {
-                workValidationContext.visitWarning(null, propertyName, "has no normalization specified. Properties of cacheable transforms must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath");
+                validationContext.visitPropertyProblem(WARNING,
```

Comment:
I suppose the warnings emitted by this method and in `fingerprintParameters` should be moved somewhere else, since they are the same for tasks.

Reply:
Yeah, we can move these now. I'd do that as a separate PR, though.

Reply:
Actually, we only forbid `ABSOLUTE` path sensitivity for cacheable _transforms_ but not for _tasks._ We can change that, but we should really do that as a separate story.

Reply:
Yeah, the warning about `ABSOLUTE` paths can stay for cacheable transforms only, the warning about no path sensitivity specified is for all units of work.